File: programmers/level1/체육복.py
import collections

def solution(n, lost, reserve):
    dic = collections.defaultdict(int)
    for i in range(len(lost)):
        dic[lost[i]-1] -= 1
    for i in range(len(reserve)):
        dic[reserve[i]-1] += 2
    total = 0
    for i in range(n):
        if dic[i]==-1:#i가 없으면
            if 0<i and dic[i-1]>1:
                dic[i-1]-=1
                dic[i]+=1
                total += 1
            elif i<n and dic[i+1]>1:
                dic[i+1]-=1
                dic[i]+=1
                total += 1
        else: total += 1
    return total

File: programmers/level1/체육복.java
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] clothes = new int[n];
        
        for(int lost_n: lost) clothes[lost_n-1]--;
        for(int reserve_n: reserve) clothes[reserve_n-1]++;
       
        for(int i=0; i<n; i++) {
            if(clothes[i] == -1) {
                if(i>0 && clothes[i-1]==1) {                    clothes[i]++;
                    clothes[i-1]--;
                } else if(i<n-1 && clothes[i+1]==1) {
                    clothes[i]++;
                    clothes[i+1]--;
                }
            }
        }
        
        int answer = 0;
        for(int i=0; i<n; i++) {
            if(clothes[i]>= 0) answer++;
        }
        return answer;
    }
}